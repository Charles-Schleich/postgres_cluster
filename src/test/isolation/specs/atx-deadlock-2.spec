setup
{
  create table t(a int, b int);
  insert into t values (1, 1), (2, 2);
}

teardown
{
  drop table t;
}

session "s1"
setup         { begin; SET deadlock_timeout = '10s'; }
step "s1u1"   { update t set b = 11 where a = 1; }
step "s1atxb" { begin autonomous; }
step "s1u2"   { update t set b = 21 where a = 2; }
step "s1atxc" { commit; }
step "s1c"    { commit; }

session "s2"
setup       { begin; SET deadlock_timeout = '10s'; }
step "s2u2" { update t set b = 22 where a = 2; }
step "s2u1" { update t set b = 11 where a = 1; }
step "s2c"  { commit; }

permutation "s1u1" "s2u2" "s1atxb" "s2u1" "s1u2" "s1atxc" "s1c" "s2c"
