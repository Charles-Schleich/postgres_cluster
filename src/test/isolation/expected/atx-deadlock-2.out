Parsed test spec with 2 sessions

starting permutation: s1u1 s2u2 s1atxb s2u1 s1u2 s1atxc s1c s2c
step s1u1: update t set b = 11 where a = 1;
step s2u2: update t set b = 22 where a = 2;
step s1atxb: begin autonomous;
step s2u1: update t set b = 11 where a = 1; <waiting ...>
step s1u2: update t set b = 21 where a = 2; <waiting ...>
step s1u2: <... completed>
step s2u1: <... completed>
error in steps s1u2 s2u1: ERROR:  deadlock detected
step s1atxc: commit;
step s1c: commit;
step s2c: commit;
